### ISSUE TITLE
Intrepid2: Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests_MPI_1 unit test failure with cuda/11.2 on Power9+Volta70 with UVM=ON and c++17

### ISSUE THREAD

**Issue by ndellingwood** [trilinos/Trilinos#11083]
## Bug Report
@trilinos/intrepid2 

### Description
The `Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests_MPI_1` unit test fails with cuda/11.2 on Power9+Volta70 with UVM=ON and c++17. This occurs with `CUDA_MANAGED_FORCE_DEVICE_ALLOC` and `CUDA_LAUNCH_BLOCKING` set or unset.

Snip of test failure:
```
...
 79: (0.00839478,0.728492,-0.90618)
 At location (0,38) actual value != expected value (1 != 1); diff is 2.12019e-12
 ** Testing operator Value **
 At location (0,38) actual value != expected value (1 != 1); diff is 2.12019e-12
 [FAILED]  (0.086 sec) BasisEquivalence_WedgeNodalVersusHierarchical_HVOL_UnitTest
 Location: /ascldap/users/ndellin/trilinos/Trilinos-pristine/packages/intrepid2/unit-test/MonolithicExecutable/BasisEquivalenceTests_Wedge.cpp:241
...
```

### Steps to Reproduce
1. SHA1: eb0988078d8bb6f6ba43e56f5caa10dc7c2ffc6b
2. Configure script:  weaver rhel8 queue
```bash
module load git cmake/3.21.2 cuda/11.2.2/gcc/8.3.1 openmpi/4.1.1/gcc/8.3.1/cuda/11.2.2 openblas/0.3.18/gcc/8.3.1
module load metis/5.1.0/gcc/8.3.1 hdf5/1.10.7/gcc/8.3.1/openmpi/4.1.1 parmetis/4.0.3/gcc/8.3.1/openmpi/4.1.1 zlib/1.2.11/gcc/8.3.1 boost/1.70.0/gcc/8.3.1
export OMPI_CXX=$TRILINOS_DIR/packages/kokkos/bin/nvcc_wrapper

cmake \
 -D CMAKE_CXX_COMPILER=`which mpicxx` \
 -D CMAKE_CXX_FLAGS="-g" \
 -D CMAKE_C_COMPILER=`which mpicc` \
 -D CMAKE_C_FLAGS="-g" \
 -D CMAKE_Fortran_COMPILER=`which mpifort` \
 -D CMAKE_Fortran_FLAGS="-g" \
 -D CMAKE_BUILD_TYPE:STRING=RELEASE \
 -D CMAKE_CXX_STANDARD=17 \
 -D BUILD_SHARED_LIBS=OFF \
\
 -D TPL_ENABLE_CUDA:STRING=ON \
 -D TPL_ENABLE_MPI:STRING=ON \
 -D MPI_BASE_DIR:PATH="$OPENMPI_ROOT" \
 -D MPI_BIN_DIR:PATH="$OPENMPI_BIN" \
 -D MPI_EXEC_POST_NUMPROCS_FLAGS:STRING="-map-by;socket:PE=4" \
\
-D TPL_ENABLE_BLAS:STRING=ON \
  -D BLAS_LIBRARY_DIRS:FILEPATH="${OPENBLAS_ROOT}/lib" \
  -D BLAS_LIBRARY_NAMES:STRING="openblas" \
-D TPL_ENABLE_LAPACK:STRING=ON \
  -D LAPACK_INCLUDE_DIRS:FILEPATH="${OPENBLAS_ROOT}/include" \
  -D LAPACK_LIBRARY_DIRS:FILEPATH="${OPENBLAS_ROOT}/lib" \
  -D LAPACK_LIBRARY_NAMES:STRING="openblas" \
-D TPL_ENABLE_Boost=ON \
  -D Boost_INCLUDE_DIRS:PATH="${BOOST_ROOT}/include" \
  -D Boost_LIBRARY_DIRS:PATH="${BOOST_ROOT}/lib" \
-D TPL_ENABLE_BoostLib=ON \
  -D BoostLib_INCLUDE_DIRS:PATH="${BOOST_ROOT}/include" \
  -D BoostLib_LIBRARY_DIRS:PATH="${BOOST_ROOT}/lib" \
-D TPL_ENABLE_Netcdf=OFF \
\
 -D Trilinos_ENABLE_COMPLEX=ON \
 -D Trilinos_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_EXAMPLES=OFF \
\
 -D Trilinos_ENABLE_Kokkos=ON \
 -D Kokkos_ENABLE_CUDA=ON \
 -D Kokkos_ENABLE_CUDA_LAMBDA=ON \
 -D Kokkos_ENABLE_CUDA_UVM=ON \
 -D Kokkos_ARCH_VOLTA70=ON \
 -D Kokkos_ARCH_POWER9=ON \
 -D Kokkos_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_KokkosKernels=ON \
 -D KokkosKernels_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Tpetra=ON \
 -D Tpetra_ENABLE_CUDA=ON \
 -D Tpetra_ENABLE_SERIAL=ON \
 -D Tpetra_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Sacado=ON \
 -D Sacado_ENABLE_TESTS=ON \
\
 -D Trilinos_ENABLE_Phalanx=ON \
 -D Phalanx_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Panzer=ON \
 -D Panzer_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Adelus=ON \
 -D Adelus_ENABLE_TESTS=ON \
 \
 -D Trilinos_ENABLE_Amesos2=ON \
 -D Amesos2_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Stokhos=ON \
 -D Stokhos_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Belos=ON \
 -D Belos_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Ifpack2=ON \
 -D Ifpack2_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_MueLu=ON \
 -D Trilinos_ENABLE_Intrepid2=ON \
 -D Intrepid2_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Zoltan2=ON \
 -D Zoltan2_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Zoltan2Sphynx=OFF \
 -D Trilinos_ENABLE_ShyLU_NodeTacho=OFF \
\
 -D Trilinos_ENABLE_Anasazi=ON \
 -D Trilinos_ENABLE_Teuchos=ON \
\
$TRILINOS_DIR
```

**Comment by mperego** [comment]
@ndellingwood This looks like an issue with tolerances being too tight. Namely we should set the two tolerances
https://github.com/trilinos/Trilinos/blob/develop/packages/intrepid2/unit-test/MonolithicExecutable/BasisEquivalenceTests_Wedge.cpp#L248-L249
to something like 1e-11.
@CamelliaDPG do you agree?

**Comment by CamelliaDPG** [comment]
@mperego, yes, I agree.

**Comment by mperego** [comment]
This should have been fixed by PR #11114 . Please reopen if you still observe this failure.
